src/advanced/components/ui/layout/TabLayout.tsx

```diff
@@ -0,0 +1,30 @@
+interface TabLayoutProps {
+  activeTab: 'products' | 'coupons';
+  handleActiveTab: (value: 'products' | 'coupons') => void;
+}
+export const TabLayout = ({ activeTab, handleActiveTab }: TabLayoutProps) => {
+  return (
+    <nav className='-mb-px flex space-x-8'>
+      <button
+        onClick={() => handleActiveTab('products')}
+        className={`py-2 px-1 border-b-2 font-medium text-sm transition-colors ${
+          activeTab === 'products'
+            ? 'border-gray-900 text-gray-900'
+            : 'border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300'
+        }`}
+      >
+        상품 관리
+      </button>
+      <button
+        onClick={() => handleActiveTab('coupons')}
+        className={`py-2 px-1 border-b-2 font-medium text-sm transition-colors ${
+          activeTab === 'coupons'
+            ? 'border-gray-900 text-gray-900'
+            : 'border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300'
+        }`}
+      >
+        쿠폰 관리
+      </button>
+    </nav>
+  );
+};
```

Comment:
```suggestion
export function AdminTab({
  tabs,
  activeTab,
  setActiveTab,
}: {
  tabs: {
    label: string;
    value: "products" | "coupons";
  }[];
  activeTab: "products" | "coupons";
  setActiveTab: React.Dispatch<React.SetStateAction<"products" | "coupons">>;
}) {
  return (
    <div className="border-b border-gray-200 mb-6">
      <nav className="-mb-px flex space-x-8">
        {tabs.map((tab) => (
          <button
            key={tab.value}
            onClick={() => setActiveTab(tab.value)}
            className={`py-2 px-1 border-b-2 font-medium text-sm transition-colors ${
              activeTab === tab.value
                ? "border-gray-900 text-gray-900"
                : "border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300"
            }`}
          >
            {tab.label}
          </button>
        ))}
      </nav>
    </div>
  );
}
```

저는 이렇게 짰는데 이런 구조는 어떠신가요? 탭의 정보를 외부에서 받는 것이 좀 더 확장성 있는 구조 같다는 생각이 들어서요!